homeassistant/components/ozw/light.py

```diff
@@ -193,12 +203,26 @@ async def async_turn_on(self, **kwargs):
                 rgbw = f"#00000000{white:02x}"
 
         elif color_temp is not None:
+            # Update color temp limits.
+            if self.values.min_kelvin:
+                self._max_mireds = color_util.color_temperature_kelvin_to_mired(
```

Comment:
The update should be done when we get feedback from the device and not in the service actuator method, here `async_turn_on`. So move this to `_calculate_rgb_values`.

Reply:
Also, we should consider renaming that method to eg `_calculate_color_values`.

Reply:
I was in the middle of re-pushing it, heh.  I caught that right after the last push.